### PR TITLE
Expose mul & div

### PR DESCRIPTION
**Pull request by 0xgleb**
## Motivation

We need to use Solidity Float code in Rust

## Solution

Expose `mul` and `div` methods

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs
- [x] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

- **New Features**
	- Added support for multiplication and division operations with the Float type.
- **Bug Fixes**
	- Improved error handling for parsing errors, exponent overflows, and division by zero.
- **Tests**
	- Expanded test coverage for multiplication, division, and various edge cases, including overflow, underflow, and error scenarios.
- **Documentation**
	- Updated documentation to clarify decimal handling in formatting.

### CODE REVIEW

File: src/concrete/DecimalFloat.sol [hunk 1]
```diff
@@ -171,4 +171,13 @@ contract DecimalFloat {
...
     }
+
+    /// Exposes `LibDecimalFloat.unpack` for offchain use.
+    /// @param float The float to unpack.
+    /// @return coefficient The coefficient of the float.
+    /// @return exponent The exponent of the float.
+    function unpack(Float float) external pure returns (int224, int32) {
+        (int256 coefficient, int256 exponent) = LibDecimalFloat.unpack(float);
+        return (int224(coefficient), int32(exponent));
+    }
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply src/concrete/DecimalFloat.sol]: `Float` is designed to be a 224 bit signed coefficient with 32 bit signed exponent. It's represented as 32 bytes of data. See `LibDecimalFloat.sol` for details

File: crates/float/src/lib.rs [hunk 2]
```diff
@@ -150,6 +149,27 @@ impl Float {
...
+            Ok((coefficient, exponent))
+        })
+    }
+
+    #[cfg(test)]
+    fn show_unpacked(self) -> Result<String, FloatError> {
+        let (coefficient, exponent) = self.unpack()?;
+        Ok(format!("{coefficient}e{exponent}"))
+    }
+
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply crates/float/src/lib.rs]: They're literally marked as `fn`

File: crates/float/src/lib.rs [hunk 3]
```diff
@@ -177,6 +198,39 @@ impl Float {
...
+    pub fn gt(self, b: Self) -> Result<bool, FloatError> {
+        let Float(a) = self;
+        let Float(b) = b;
+        let calldata = DecimalFloat::gtCall { a, b }.abi_encode();
+
+        execute_call(Bytes::from(calldata), |output| {
+            let decoded = DecimalFloat::gtCall::abi_decode_returns(output.as_ref())?;
+            Ok(decoded)
+        })
+    }
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply crates/float/src/lib.rs]: The largest duplication was refactored into `execute_call`. The rest is not significant

File: crates/float/src/lib.rs [hunk 4]
```diff
@@ -209,9 +263,40 @@ impl Sub for Float {
...
+        let Float(a) = self;
+        let Float(b) = b;
+        let calldata = DecimalFloat::divCall { a, b }.abi_encode();
+
+        execute_call(Bytes::from(calldata), |output| {
+            let decoded = DecimalFloat::divCall::abi_decode_returns(output.as_ref())?;
+            Ok(Float(decoded))
+        })
+    }
+}
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply crates/float/src/lib.rs:294]: All `Float` operations are `Result`s and the compiler would notify the user of that being the case if they try to use the outputs in other ways

**Review by hardyjosh — APPROVED** [review]
